hibernate-reactive-core/src/main/java/org/hibernate/reactive/sql/results/spi/ReactiveSingleResultConsumer.java

```diff
@@ -29,11 +30,14 @@ public CompletionStage<T> consume(
 		return rowProcessingState.next()
 				.thenCompose( hasNext -> rowReader
 						.reactiveReadRow( rowProcessingState, processingOptions )
-						.thenApply( result -> {
+						.thenCompose( result -> {
 							rowProcessingState.finishRowProcessing( true );
 							rowReader.finishUp( rowProcessingState );
-							jdbcValuesSourceProcessingState.finishUp( false );
-							return result;
+							session.getPersistenceContext();
+							jdbcValuesSourceProcessingState.finishLoadingCollections();
+							return ( (ReactivePersistenceContextAdapter) session.getPersistenceContextInternal() )
+									.reactivePostLoad( jdbcValuesSourceProcessingState, null )
```

Comment:
The code we had before was more similar to the one in ORM.
For example, ORM doesn't call `.finishLoadingCollections`.

I'm not saying that this is wrong, but could you explain a bit what's going on?

Reply:
in ORM the JdbcValuesSourceProcessingStateStandardImpl  finishUp methos is 
```
public void finishUp(boolean registerSubselects) {
		// now we can finalize loading collections
		finishLoadingCollections();

		getSession().getPersistenceContextInternal()
				.postLoad( this,
						registerSubselects ? executionContext::registerLoadingEntityHolder : null );
	}
```

Reply:
Got it, thanks